core/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java

```diff
@@ -0,0 +1,188 @@
+
+    // URL can not determine offset and limit, let javascript in jsp determine and redirect
+    if (request.getParameter("offset") == null && request.getParameter("limit") == null) {
+      getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request, response);
+      return;
+    }
+
+    try {
+      int offset = Integer.parseInt(request.getParameter("offset"));
+      int limit = Integer.parseInt(request.getParameter("limit"));
+      List<Integer> subFileIds = fileIds.subList(offset, offset + limit);
+      List<UiFileInfo> uiFileInfos = new ArrayList<UiFileInfo>(subFileIds.size());
+      for (int fileId : subFileIds) {
+        uiFileInfos.add(getUiFileInfo(tachyonClient, fileId, TachyonURI.EMPTY_URI));
+      }
+      request.setAttribute("fileInfos", uiFileInfos);
+    } catch (FileDoesNotExistException fdne) {
+      request.setAttribute("fatalError", "Error: Invalid FileId " + fdne.getMessage());
+      getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request,
+          response);
+      return;
+    } catch (NumberFormatException nfe) {
+      request.setAttribute("fatalError",
+          "Error: offset or limit parse error, " + nfe.getLocalizedMessage());
+      getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request, response);
+      return;
+    } catch (IndexOutOfBoundsException iobe) {
+      request.setAttribute("fatalError", "Error: offset or offset + limit is out of bound, "
+          + iobe.getLocalizedMessage());
+      getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request, response);
+      return;
+    } catch (IllegalArgumentException iae) {
+      request.setAttribute("fatalError", iae.getLocalizedMessage());
+      getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request, response);
+      return;
+    }
+
+    getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request, response);
+  }
+
+  private List<Integer> getFileIds(TachyonFS tachyonClient) {
+    Set<Integer> fileIds = new HashSet<Integer>();
+    for (StorageDir storageDir : mWorkerStorage.getStorageDirs()) {
+      for (long blockId : storageDir.getBlockIds()) {
+        int fileId = BlockInfo.computeInodeId(blockId);
+        fileIds.add(fileId);
+      }
+    }
+    List<Integer> sortedFileIds = new ArrayList<Integer>(fileIds);
+    Collections.sort(sortedFileIds);
+    return sortedFileIds;
+  }
+
+  /**
+   * Gets the uiFileInfo object that represents the fileId, or the filePath if fileId is -1.
+   *
+   * @param tachyonClient the TachyonFS client.
+   * @param fileId the file id of the file.
+   * @param filePath the path of the file. valid iff fileId is -1.
+   * @return the UiFileInfo object of the file.
+   * @throws FileDoesNotExistException
+   * @throws IOException
+   */
+  private UiFileInfo getUiFileInfo(TachyonFS tachyonClient, int fileId, TachyonURI filePath)
```

Comment:
Yes, I was following the same code pattern of TachyonFS.getFileStatus. I now added anthoter two signatures that call this one.